test/e2e_node/runtime_conformance_test.go

```diff
@@ -0,0 +1,138 @@
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	retryTimeout = time.Minute * 4
+	pollInterval = time.Second * 5
+)
+
+var _ = Describe("Container Runtime Conformance Test", func() {
+	var cl *client.Client
+
+	BeforeEach(func() {
+		// Setup the apiserver client
+		cl = client.NewOrDie(&restclient.Config{Host: *apiServerAddress})
+	})
+
+	Describe("container conformance blackbox test", func() {
+		Context("when running a container that terminates", func() {
+			var containerCase ConformanceContainer
+
+			BeforeEach(func() {
+				containerCase = ConformanceContainer{
+					Container: api.Container{
+						Image:           "gcr.io/google_containers/busybox",
+						Name:            "busybox",
+						Command:         []string{"sh", "-c", "env"},
+						ImagePullPolicy: api.PullIfNotPresent,
+					},
+					Client:   cl,
+					Phase:    api.PodSucceeded,
+					NodeName: *nodeName,
+				}
+				err := containerCase.Create()
+				Expect(err).NotTo(HaveOccurred())
+
+				//Waiting for the 'running' phase
+				for start := time.Now(); time.Since(start) < retryTimeout; time.Sleep(pollInterval) {
+					if container, err := containerCase.Get(); err == nil && container.Phase == api.PodRunning {
+						break
+					}
+				}
+			})
+
+			It("it should report its phase as 'succeeded' and get a same container [Conformance]", func() {
+				var container ConformanceContainer
+				var err error
+				Eventually(func() api.PodPhase {
+					container, err = containerCase.Get()
+					return container.Phase
+				}, retryTimeout, pollInterval).Should(Equal(api.PodSucceeded))
+				Expect(err).NotTo(HaveOccurred())
+				Expect(container).Should(CContainerEqual(containerCase))
+			})
+			It("it should be possible to delete [Conformance]", func() {
+				err := containerCase.Delete()
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			AfterEach(func() {
+				containerCase.Delete()
+				Eventually(func() error {
+					_, err := containerCase.Get()
+					return err
+				}, retryTimeout, pollInterval).Should(HaveOccurred())
+			})
+
+		})
+		Context("when running a container with invalid image", func() {
+			var containerCase ConformanceContainer
+			BeforeEach(func() {
+				containerCase = ConformanceContainer{
+					Container: api.Container{
+						Image:           "foo.com/foo/foo",
+						Name:            "foo",
+						Command:         []string{"foo", "'Should not work'"},
+						ImagePullPolicy: api.PullIfNotPresent,
+					},
+					Client:   cl,
+					Phase:    api.PodPending,
+					NodeName: *nodeName,
+				}
+				err := containerCase.Create()
+				Expect(err).NotTo(HaveOccurred())
+
+				//Waiting for the 'running' phase
+				for start := time.Now(); time.Since(start) < retryTimeout; time.Sleep(pollInterval) {
+					if container, err := containerCase.Get(); err == nil && container.Phase == api.PodRunning {
```

Comment:
My previous assumption is it will be a 'bug' if it became 'running'.
I'll change it to ' container.Phase != api.PodPending' .